### PR TITLE
Tob updates

### PR DESCRIPTION
**Pull request by symposion**
Hey Mark,

There were a bunch of problems with the first cut of the TOB stuff which were pointed out to me. Have put in some fixes here + unfortunately I've had to make a breaking change to the JSON handling - it affects only 4 entries from the MM, but that's enough to cause it to barf on everyone's monster files. As a result I've had to bump the version number of all the JSON files, and the script now gives a clear error message instructing people to redownload their monster files.

Cheers,

### CODE REVIEW

**Comment by mlenser** [comment]
The \n is there for the line break - it shouldn't be removed.

**Comment by symposion** [comment]
Ok, then I need another solution for working with lair actions and regional
effects from the statblock. At the moment I'm assuming that a regional
effect/lair action appears on a single line and is ended by a linebreak.
The rules that define what is allowable within a particular section of the
statblock are the same rules that validate the JSON input. Removing the
'\n' from the 4 dragons' actions seemed like an easy solution and I was
being lazy but if you feel strongly about it I can change the definition to
say that each lair action/regional effect must start with either a unicode
bullet character or an asterisk. I was a bit wary of this because bullets
sometimes don't make it through the text translation, but I think it makes
for a better and more robust format specification anyway, so I'll change it
to that...

**Comment by mlenser** [comment]
There are 545 places in the monster manual alone where "\n" is used.
A regional effect and lair action are parsed into the same field as "text" from traits or an action. So line breaks should be allowed.

Perhaps I'm not seeing what is different here?

**Comment by symposion** [comment]
The difference is that when parsing from a text statblock (which shares a
lot of code with the JSON import validation), I need some way to demarcate
the end of each regional/lair action. Unlike traits and actions they don't
have a neat little title case name followed by a full stop at the beginning
of each one. Previously the statblock import basically didn't work with
lair actions and regional effects, but for TOB I wanted to make it work
because there are a lot and I didn't want to have to do them manually. I
assumed that all lair actions and regional effects appeared on a single
line - which they do, except for Green and White dragons, which have one
each that span more than one line.

Anyway, it's fine, I've changed the parser to look for a bullet or a start
to mark the beginning of an action/effect. This fits with the standard
format for MM and TOB, so I think it's reasonable.

Probably the part which is not making sense to you is why the JSON import
(which has structure to demarcate this stuff) should be affected by the
vagaries of the text parsing. That's because I wanted to share validation
between the two parts, so to some extent, the regex that defines what is
valid in the JSON overlaps with what defines the structure of the statblock
text, so I didn't have to do everything twice.

TL;DR; don't worry, I'm backing out the changes and will push a new,
uncontroversial commit in a moment...

Lucian

**Comment by symposion** [comment]
Ok, I've amended my original commit to reduce the amount of spam in the git log. I've reinstated the linebreaks, but I've still had to bump the version numbers because the (erroneously) empty "regionalEffectsFade" value for Adult Gold Dragon will cause the latest version of the script to barf. We should have been on 1.0.0 ages ago anyway if we're doing semver properly. Also, while I was at it, I noticed that for some reason the Ancient Gold Dragon doesn't have its lair actions or regional effects, so I copied those across from the Adult one. Hopefully this is all unproblematic now.
